contrib/QAT-ZSTD-Plugin-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,86 @@
+# Intel® QuickAssist Technology ZSTD Plugin (QAT ZSTD Plugin) is a plugin to Zstandard*(ZSTD*) for accelerating compression by QAT.
+
+if (ENABLE_QAT_OUTOFTREE)
+# Intel QATZSTD out-of-tree build
```

Comment:
Every future developer reading this build description will have no idea what the in-tree vs. the out-tree build alternatives are and why one should use one over the other.

Please add comments to the beginning of the CMake file, explaining both alternatives in detail.

Reply:
Done